CMakeLists.txt

```diff
@@ -55,6 +55,14 @@ set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+# We enable C here to make all related details and flags
```

Comment:
Sorry, but I just can't parse this comment. It's meaningless to anyone who doesn't know exactly what these python scripts do.

Could you please change this comment to explain what it does rather than why it's required?

Reply:
How about this one:
```
# A language must be enabled in the highest directory common to all
# targets that use it directly or indirectly. We use the C language in
# the test-security-check target; therefore, we enable it here.
```
?


All restrictions applied to the `enable_language()` call are listed in the CMake [docs](https://cmake.org/cmake/help/latest/command/enable_language.html). One of them:
> It must be called in the highest directory common to all targets using the named language directly for compiling sources or indirectly through link dependencies. It is simplest to enable all needed languages in the top-level directory of a project.

Reply:
I still find this very confusing. At a high level, it makes sense to me to enable C if we're going to use it later. But `test-security-check` is such an obscure reason that I still find the comment confusing.

In fact, why is that script using C rather than C++ anyway? That seems like the real issue here to me.

Reply:
> I still find this very confusing. At a high level, it makes sense to me to enable C if we're going to use it later. But `test-security-check` is such an obscure reason that I still find the comment confusing.
> 
> 
> 
> In fact, why is that script using C rather than C++ anyway? That seems like the real issue here to me.

I totally agree with the last paragraph. Initially, I had a patch was that. But it appears too invasive; therefore, I decided to do not touch that code during migration to CMake.

Reply:
I think it makes sense to go ahead and upstream that.
1. In those scripts we should be testing the language we're actually using (we may need to add more exceptions for c++ deps/symbols along the way)
2. We can discuss the C language issue here without this crazy dependency getting in the way

Reply:
Untested branch here: https://github.com/theuni/bitcoin/commits/cxx-security-check/

Lief is broken for me atm for some reason, so I'm unable to test if that's enough.

Reply:
> Untested branch here: https://github.com/theuni/bitcoin/commits/cxx-security-check/

Thanks! [Taken](https://github.com/hebasto/bitcoin/pull/252#issuecomment-2207356538).